Modules/System/Azure Blob Services API/src/ABSBlobClient.Codeunit.al

```diff
@@ -621,6 +621,54 @@ codeunit 9053 "ABS Blob Client"
         exit(ABSClientImpl.GetBlobTags(BlobName, Tags, OptionalParameters))
     end;
 
+    /// <summary>
+    /// The Get Blob Properties operation returns all user-defined metadata, standard HTTP properties, and system properties for the blob. It doesn't return the content of the blob.
+    /// see: https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties?tabs=azure-ad
+    /// </summary>
+    /// <param name="BlobName">The name of the blob.</param>
+    /// <returns>An operation response object. The Blob properties are in the response headers.</returns>
+    procedure GetBlobProperties(BlobName: Text): Codeunit "ABS Operation Response"
```

Comment:
I really like this addition.
could you maybe add the other optional request header parameter from the documentation you mention in the summary? I think then it would be more complete.

_x-ms-upn_: Optional. Version 2020-06-12 and later. Valid for accounts with hierarchical namespace enabled. If the user identity values that are returned in the x-ms-owner are true, they're transformed from Azure Active Directory object IDs to user principal names. If the values are false, they're returned as Azure Active Directory object IDs. The default value is false. Note that group and application object IDs are not translated, because they don't have unique friendly names.
https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties?tabs=azure-ad#request-headers

Reply:
Added the request header x-ms-upn with value true.

Reply:
@ajkauffmann I'm sorry, I think my comment wasn't very clear, my bad.

I thought of something like the following for the `codeunit 9047 "ABS Optional Parameters"`:

```al
    /// <summary>
    /// Sets the value for 'x-ms-upn' HttpHeader for a request.
    /// Version 2020-06-12 and later.
    /// Valid for accounts with hierarchical namespace enabled.
    /// If the user identity values that are returned in the x-ms-owner are true, they're transformed from Azure Active Directory object IDs to user principal names.
    /// If the values are false, they're returned as Azure Active Directory object IDs. The default value is false.
    /// Note that group and application object IDs are not translated, because they don't have unique friendly names.
    /// </summary>
    /// <param name="Value">Boolean value specifying the HttpHeader value</param>
    procedure ReturnUserPrincipalName("Value": Boolean)
    var
        ValueText: Text;
    begin
        // Set as text, because otherwise it might give different formatted values based on language locale
        if "Value" then
            ValueText := 'true'
        else
            ValueText := 'false';

        SetRequestHeader('x-ms-upn', ValueText);
    end;
```